docs/source/models.rst

```diff
@@ -3655,23 +3655,113 @@ Evaluating with Extra Metrics
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 If the default set of metrics is insufficient, you can supply ``extra_metrics`` and ``custom_artifacts``
-to :py:func:`mlflow.evaluate()` to produce custom metrics and artifacts for the model(s) that you're evaluating.
+to :py:func:`mlflow.evaluate()` to produce extra metrics and artifacts for the model(s) that you're evaluating.
+
+To define an extra metric, you should define an ``eval_fn`` function that takes in ``predictions``, ``targets``,
+and ``metrics`` as inputs and outputs a ``MetricValue`` object. 
+
+The ``MetricValue`` class has three attributes:
+
+* ``scores``: used for per-row metrics 
+* ``aggregate_results``: a dictionary that maps metric names to their aggregate results
+* ``justification``: a per-row justification of the values in ``scores``
+The code block below demonstrates how to define a custom metric evaluation function:
+
+.. code-block:: python
+    from mlflow.metrics import MetricValue
+
+    def my_metric_eval_fn(predictions, targets, metrics):
+        scores = predictions
+        mymetric = np.sum(np.abs(predictions - targets))
+        return MetricValue(scores=scores, aggregate_results={"mymetric": mymetric})
+
+Once you have defined an ``eval_fn``, you then use ``make_metric()`` to wrap this ``eval_fn`` function into a metric.
+In addition to ``eval_fn``, ``make_metric()`` requires an additional parameter , ``greater_is_better``, for optimization purposes. This parameter
+indicates whether this is a metric we want to maximize or minimize.
+
+.. code-block:: python
+    from mlflow.metrics import make_metric
+
+    mymetric = make_metric(eval_fn=my_metric_eval_fn, greater_is_better=False)
+
+The extra metric allows you to either evaluate a model directly, or to evaluate an output dataframe. 
+
+To evaluate the model directly, you will have to provide ``mlflow.evaluate()`` either a pyfunc model
+instance, a URI referring to a pyfunc model, or a callable function that takes in the data as input 
+and outputs the predictions.
+
+.. code-block:: python
+    def model(x):
+        return x["inputs"]
+
+    eval_dataset = pd.DataFrame({"targets": [1,2,3,4,5,6,7,8], "inputs": [1,2,3,4,5,6,7,8]})
+
+    mlflow.evaluate(model, eval_dataset, targets="targets", extra_metrics=[mymetric])
+
+To directly evaluate an output dataframe, you can **omit** the ``model`` parameter. However, you will need
+ to set the ``predictions`` parameter in ``mlflow.evaluate()`` in order to evaluate an inference output dataframe. 
+
+.. code-block:: python
+
+    eval_dataset = pd.DataFrame({"targets": [1,2,3,4,5,6,7,8], "predictions": [1,2,3,4,5,6,7,8]})
```

Comment:
I haven't tested the case when there is no "input" column. Does it work? I would expect to get some errors if the `data` has no input column (even though no input is needed in this case). It would be awesome if it already works.

Reply:
yep the code I added has been tested on dogfood